src/Nethermind/Nethermind.Merge.Plugin/Handlers/PayloadService.cs

```diff
@@ -174,31 +175,25 @@ private async Task ProduceIdealBlock(ulong payloadId, BlockHeader parentHeader,
             return null;
         }
 
-        private ulong RentNextPayloadId()
+        private byte[] ComputeNextPayloadId(BlockHeader parentHeader, PayloadAttributes payloadAttributes)
         {
-            lock (_locker)
-            {
-                while (_payloadStorage.ContainsKey(_currentPayloadId))
-                {
-                    if (_currentPayloadId == ulong.MaxValue)
-                        _currentPayloadId = 1;
-                    else
-                        ++_currentPayloadId;
-                }
-
-                ulong rentedPayloadId = _currentPayloadId;
-                ++_currentPayloadId;
-                return rentedPayloadId;
-            }
+            byte[] input = new byte[32 + 32 + 32 + 20];
```

Comment:
Take a look here: https://github.com/ethereum/execution-apis/blob/89070c4d6f2fbe125a906f0cca1fedda84b8a2a6/src/engine/specification.md#engine_getpayloadv1

PayloadId should be data[8]. I've already made it on the main kintsugi branch. You need to adjust it only.

Reply:
yes this function returns the first 8 bytes of the hash. 

You have turned the `payloadId` to a hex string on the main branch. Are we using sequential IDs?